### PR TITLE
Fixup command validation

### PR DESCRIPTION
**Pull request by yenda**
fix validation of commands

status: ready

### CODE REVIEW

**Review by janherich — APPROVED** [review]

File: src/status_im/transport/db.cljs [hunk 1]
```diff
@@ -51,12 +51,7 @@
 (spec/def :message.content/command-path (spec/tuple string? (spec/coll-of (spec/or :scope keyword? :chat-id string?) :kind set? :min-count 1)))
 (spec/def :message.content/params (spec/map-of keyword? any?))
 
-(spec/def ::content (spec/conformer (fn [content]
```
Comment (yenda) [inline src/status_im/transport/db.cljs:54]: this is unused

File: src/status_im/transport/db.cljs [hunk 2]
```diff
@@ -51,12 +51,7 @@
...
-                                      (cond
-                                        (string? content) {:text content}
-                                        (map? content) content
-                                        :else :clojure.spec/invalid))))
-(spec/def ::content-type #{"text/plain" "command"})
```
Comment (yenda) [inline src/status_im/transport/db.cljs:59]: legacy "command-request" was missing, so validation was filtering incoming request commands

File: src/status_im/transport/db.cljs [hunk 3]
```diff
@@ -91,6 +86,10 @@
   (spec/merge :message/message-common
               (spec/keys :req-un [:message.command/content])))
 
+(defmethod content-type "command-request" [_]
```
Comment (yenda) [inline src/status_im/transport/db.cljs:89]: when we receive a legacy content-type "command-request", we validate it the same as content-type "command"

File: src/status_im/transport/message/v1/protocol.cljs [hunk 4]
```diff
@@ -111,7 +111,7 @@
              :js-obj     (:js-obj cofx))]})
   (validate [this]
     (when (spec/valid? :message/message this)
-      (spec/conform :message/message this))))
+      this)))
```
Comment (yenda) [inline src/status_im/transport/message/v1/protocol.cljs:114]: do not conform the message, it is already valid and conforming would introduce weird spec specific constructs.

**Review by jakubgs — APPROVED** [review]
I'm don't fully understand what is happening here, but I get that it's supposed to undo some of the changes done in https://github.com/status-im/status-react/commit/81b2afebc127deb3a6bef27091f13ea0fdcca6e8 that introduced a bug. So I'm approving it. Worst case it won't fix what is already broken.

**Pull request merged**